cep-0016-2.md

```diff
@@ -0,0 +1,86 @@
+> * (Typed) repodata parsers would need to be updated to handle the new field.
+We propose that these reasons no longer hold with [sharded repodata](https://github.com/conda/ceps/blob/main/cep-0016.md).
+
+**It would require extending the repodata schema, currently not formally standardized.**
+
+We propose to add a `run_export` field to each record that mimics the specification from CEP-12.
+
+If the `run_export` field is not present in the record it means no `run_export` information is stored with the record, and a fallback mechanism should be used to acquire the run-export information.
+
+Since *adding* a field will not break existing parsers we feel this is safe and does not require a schema change.
+
+**It would increase the size of the already heavy repodata.json files.**
+
+The size of the shards would grow if run-exports are added.
+
+Let's take a look at the current sizes of run_exports.json and repodata.json files.
+
+| channel + subdir | size of repo_data.json | size of run_exports.json | repodata.json.zst | run_exports.json.zst |
+|------------------|------------------------|--------------------------|--------------------------|-----------------------------|
+| conda-forge + linux-64 | 254 MB | 34.7 MB (11%) | 38.4 MB | 2.2MB (5%) |
+| conda-forge + noarch | 107 MB | 13.6 MB (11%) | 16.7 MB | 0.9 MB (5%) |
+| conda-forge + osx-arm64 | 99.8 MB | 11.8 MB (11%) | 12.6 MB | 0.8 MB (6%) |
+| conda-forge + win-64 | 185 MB | 22.1 MB (11%) | 24.7 MB | 1.4 MB (5%) |
+
+Since the repodata shards are also compressed we can conclude that in practice adding run exports information would increase the size of the repodata shards by roughly 5-6%.
+
+With the introduction of sharded repodata in [CEP-16](https://github.com/conda/ceps/blob/main/cep-0016.md) the issues with size (and scale) have been effectively mitigated. Adding 5-6% to the total size of the shards will not pose a risk since all advantages of sharded repodata mentioned in the CEP still hold.
+
+**(Typed) repodata parsers would need to be updated to handle the new field.**
+
+Unless parsers are very strict about unknown fields (which was not a requirement for sharded repodata shards) this will not pose an issue. Since the absence of the field means that the state of the run-exports is unknown adding the field does not require a schema change.
+
+## Patching
+
+With the run-exports part of the repodata we propose to also allow repodata patching these fields. The original run-exports can still be extracted from the packages. We do not see a reason how the run-export information is different from other patchable information stored in the repodata (like the dependencies).
+
+To facilitate these patches, the `patch_instructions_version` in `patch_instructions.json` files is incremented to `2`. A `patch_instructions_version: 2` file CAN contain a `run_exports: ...` field in the patch instructions that MUST be used to patch generated `run_exports.json` files.
```

Comment:
Minor style note: standard RFC 2119 language uses "MAY" rather than "CAN"; as in:

> A `patch_instructions_version: 2` file MAY contain a ...